src/Services/SearchService.php

```diff
@@ -364,4 +367,31 @@ private function getVariationIdForRedirect(string $query, array $documents)
 
         return $mainVariationId;
     }
+
+    /**
+     * @see \IO\Extensions\Filters\URLFilter::buildItemURL (source)
+     */
+    private function buildItemURL($itemData, $withVariationId = true): string
+    {
+        $itemId = $itemData['item']['id'];
+        $variationId = $itemData['variation']['id'];
+
+        if ($itemId === null || $itemId <= 0) {
+            return '';
+        }
+
+        /** @var UrlBuilderRepositoryContract $urlBuilderRepository */
+        $urlBuilderRepository = pluginApp(UrlBuilderRepositoryContract::class);
+
+        $includeLanguage = Utils::getLang() !== Utils::getDefaultLang();
+        if ($variationId === null || $variationId <= 0) {
+            return $urlBuilderRepository->buildItemUrl($itemId)->toRelativeUrl($includeLanguage);
+        } else {
+            $url = $urlBuilderRepository->buildVariationUrl($itemId, $variationId);
+
+            return $url->append(
+                $urlBuilderRepository->getSuffix($itemId, $variationId, $withVariationId)
+            )->toRelativeUrl($includeLanguage);
```

Comment:
Really strange to use a boolean here and not a language string, but it's how plenty defines it.

Reply:
It is what it is.